src/Raven.Server/ServerWide/Commands/ClusterTransactionCommand.cs

```diff
@@ -891,20 +926,76 @@ private static unsafe SingleClusterDatabaseCommand ReadCommand<TTransaction>(Tra
 
         public override object FromRemote(object remoteResult)
         {
-            var errors = new List<ClusterTransactionErrorInfo>();
-            if (remoteResult is BlittableJsonReaderArray array)
+            if (remoteResult is BlittableJsonReaderObject bjro)
+            {
+                return GetResults(bjro);
+            }
+
+            if (remoteResult is BlittableJsonReaderArray bjra)
+            {
+                return GetErrors(bjra);
+            }
+
+            return base.FromRemote(remoteResult);
+        }
+
+        internal static ClusterTransactionResult GetResults(BlittableJsonReaderObject bjro)
+        {
+            var results = new ClusterTransactionResult();
+            if (bjro.TryGet(nameof(ClusterTransactionResult.GeneratedResult), out BlittableJsonReaderArray bjra))
+                results.GeneratedResult = GetGeneratedResults(bjra);
+
+            return results;
+        }
+
+        private static DynamicJsonArray GetGeneratedResults(BlittableJsonReaderArray bjra)
+        {
+            var array = new DynamicJsonArray();
+
+            if (bjra != null)
             {
-                foreach (var o in array)
+                foreach (BlittableJsonReaderObject item in bjra)
                 {
-                    if (o is not BlittableJsonReaderObject blittable)
-                        continue;
+                    array.Add(ToDynamicJsonValue(item));
+                }
+            }
 
-                    errors.Add(ToClusterTransactionErrorInfo(blittable));
+            return array;
+        }
+
+        private static DynamicJsonValue ToDynamicJsonValue(BlittableJsonReaderObject bjro)
```

Comment:
See if we can do:

```

    public class Result
    {
        public string Type;
        [ForceJsonSerialization("@change-Vector")]
        public string ChangeVector;
    }
    ```

Reply:
please opene an issue for that

Reply:
what is the issue number?

Reply:
[RavenDB-21984](https://issues.hibernatingrhinos.com/issue/RavenDB-21984) Turn cluster transaction data commands into objects